datalad/support/gitrepo.py

```diff
@@ -3460,11 +3482,28 @@ def save_(self, message=None, paths=None, _status=None, **kwargs):
                         logger=lgr)
 
 
+        # https://github.com/datalad/datalad/issues/6558
+        # file could have become a directory. Unfortunately git
+        # would then mistakenly refuse to commit if that old path is also
+        # given to commit, so we better filter it out
+        if status_state['deleted'] and status_state['added']:
+            # check if any "deleted" is a directory now. Then for those
+            # there should be some other path under that directory in 'added'
+            for f in [_ for _ in status_state['deleted'] if _.is_dir()]:
+                # this could potentially be expensive if lots of files become
+                # directories, but it is unlikely to happen often
+                # Note: PurePath.is_relative_to was added in 3.9 and seems slowish
+                # path_is_subpath faster, also if comparing to "in f.parents"
+                f_str = str(f)
+                if any(path_is_subpath(str(f2), f_str) for f2 in status_state['added']):
+                    status_state['deleted'].pop(f)  # do not bother giving it to commit below in _save_post
```

Comment:
Same here. I think we either need to keep it in sync or don't even bother to have `status` (or `status_state`) in  a local var.

Reply:
per my response above there is no `status` any longer -- there is nothing to keep in sync etc. But this `.pop` is the "heart" of the fix for https://github.com/datalad/datalad/issues/6558 as the [comment above suggests](https://github.com/datalad/datalad/pull/6581/files#diff-bfc1c47d032fc72b15b6a79db3d49e4c4334cf96318c5efe48a3403744b32fb1R3485).  Nothing to be done here IMHO